src/main/java/seedu/multitasky/model/ModelManager.java

```diff
@@ -138,19 +138,27 @@ public synchronized void addEntry(ReadOnlyEntry entry)
     // @@author A0126623L
     @Override
     public void changeEntryState(ReadOnlyEntry entryToChange, Entry.State newState)
-            throws DuplicateEntryException, EntryNotFoundException {
-        _entryBook.changeEntryState(entryToChange, newState);
-        indicateEntryBookChanged();
+            throws DuplicateEntryException, EntryNotFoundException, OverlappingEventException {
+        try {
+            _entryBook.changeEntryState(entryToChange, newState);
+        } finally {
+            indicateEntryBookChanged();
```

Comment:
is there a reason for separating these 2 to a try and finally block? because it shouldn't be raising a entrybookchanged event if the try block fails and the entrybook didnt actually change right?